### PR TITLE
[Logging]improve kafka certificate precheck

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
need to tell user not support use self-signed ca when enable client auth

Solution:
return error message when user try to use self-signed ca with client
auth enabled

Issue:
https://github.com/rancher/rancher/issues/18862

### CODE REVIEW

**Comment by orangedeng** [comment]
Need rebase.

**Comment by orangedeng** [comment]
LGTM

File: pkg/api/customization/logging/validator.go [hunk 1]
```diff
@@ -85,5 +86,14 @@ func validateKafka(kafkaConfig *v3.KafkaConfig) error {
...
 		return httperror.NewAPIError(httperror.InvalidBodyContent, "Plain SASL authentication requires SSL is configured")
 	}
+
+	isSelfSigned, err := loggingutils.IsSelfSigned([]byte(kafkaConfig.Certificate))
+	if err != nil {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "parse certificate failed")
+	}
+
+	if loggingutils.IsClientAuthEnaled(kafkaConfig.ClientCert, kafkaConfig.ClientKey) && isSelfSigned {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "Certificate verify failed, Kafka doesn't support self-signed certificate when client authentication is enabled")
```
Comment (alena1108) [inline pkg/api/customization/logging/validator.go]: nit: certificate verification

File: pkg/controllers/user/logging/utils/kafka.go [hunk 2]
```diff
@@ -239,6 +249,18 @@ func (w *kafkaTestWrap) kafkaConn(dial dialer.Dialer, config *tls.Config, smartH
 	return kafka.NewConn(conn, w.Topic, defaultPartition), nil
 }
 
+func (w *kafkaTestWrap) verifyKafkaCert() error {
+	isSelfSigned, err := IsSelfSigned([]byte(w.Certificate))
+	if err != nil {
+		return err
+	}
+
+	if IsClientAuthEnaled(w.ClientCert, w.ClientKey) && isSelfSigned {
```
Comment (alena1108) [inline pkg/controllers/user/logging/utils/kafka.go]: nit: putting isSelfSigned as a first check in if can help to save on calling isClientAuthEnabled when isSelfSigned is false.

File: pkg/controllers/user/logging/utils/kafka.go [hunk 3]
```diff
@@ -239,6 +249,18 @@ func (w *kafkaTestWrap) kafkaConn(dial dialer.Dialer, config *tls.Config, smartH
...
+		return errors.New("Certificate verify failed, Kafka doesn't support self-signed certificate when client authentication is enabled")
```
Comment (alena1108) [inline pkg/controllers/user/logging/utils/kafka.go]: Certificate verification

**Comment by aiwantaozi** [comment]
updated

**Comment by alena1108** [comment]
LGTM

**Pull request merged**